ballerina/http_client_payload_builder.bal

```diff
@@ -161,15 +162,15 @@ isolated function jsonPayloadBuilder(Response response, TargetType targetType) r
 isolated function nonNilablejsonPayloadBuilder(Response response, typedesc<anydata> targetType)
         returns anydata|ClientError {
     json payload = check response.getJsonPayload();
-    var result = payload.fromJsonWithType(targetType);
+    var result = jsondata:parseAsType(payload, {enableConstraintValidation: false}, targetType);
```

Comment:
Lets add a test case to validate the projection. For example if we return a json which has fields `a`, `b`, and `c` and try to data-bind it to a **closed** record with only `a` and `b`, then it should return an error since the projection is disabled by default

Reply:
Updated